endpoints/openrtb2/amp_auction.go

```diff
@@ -557,7 +554,27 @@ func setAmpExt(site *openrtb.Site, value string) {
 	}
 }
 
-func readConsent(url *url.URL) string {
+func readConsent(url *url.URL) (privacy.PolicyWriter, error) {
```

Comment:
Should this be `policyFromUrl()`? I was a bit confused and though `readConsent()` without qualifications should read consent information from the request, as that is the "normal" location for this info. `readConsentFromUrl()` could also be a possibility, but you already used that below.